addons/stock_account/stock_account.py

```diff
@@ -26,6 +26,116 @@
+            # debit account dacc will be the output account
+            # first check the product, if empty check the category
+            dacc = i_line.product_id.property_stock_account_output and i_line.product_id.property_stock_account_output.id
+            if not dacc:
+                dacc = i_line.product_id.categ_id.property_stock_account_output_categ and i_line.product_id.categ_id.property_stock_account_output_categ.id
+            # in both cases the credit account cacc will be the expense account
+            # first check the product, if empty check the category
+            cacc = i_line.product_id.property_account_expense and i_line.product_id.property_account_expense.id
+            if not cacc:
+                cacc = i_line.product_id.categ_id.property_account_expense_categ and i_line.product_id.categ_id.property_account_expense_categ.id
+            if dacc and cacc:
+                price_unit = i_line.move_id and i_line.move_id.price_unit or i_line.product_id.standard_price
+                return [
+                    {
+                        'type':'src',
+                        'name': i_line.name[:64],
+                        'price_unit':price_unit,
+                        'quantity':i_line.quantity,
+                        'price':self._get_price(cr, uid, inv, company_currency, i_line, price_unit),
+                        'account_id':dacc,
+                        'product_id':i_line.product_id.id,
+                        'uos_id':i_line.uos_id.id,
+                        'account_analytic_id': False,
+                        'taxes':i_line.invoice_line_tax_id,
+                    },
+
+                    {
+                        'type':'src',
+                        'name': i_line.name[:64],
+                        'price_unit':price_unit,
+                        'quantity':i_line.quantity,
+                        'price': -1 * self._get_price(cr, uid, inv, company_currency, i_line, price_unit),
+                        'account_id':cacc,
+                        'product_id':i_line.product_id.id,
+                        'uos_id':i_line.uos_id.id,
+                        'account_analytic_id': False,
+                        'taxes':i_line.invoice_line_tax_id,
+                    },
+                ]
+        return []
+
+class account_invoice(osv.osv):
+    _inherit = "account.invoice"
+
+    def _prepare_refund(self, cr, uid, invoice, date_invoice=None, date=None, description=None, journal_id=None, context=None):
+        invoice_data = super(account_invoice, self)._prepare_refund(cr, uid, invoice, date, date,
+                                                                    description, journal_id, context=context)
+        #for anglo-saxon accounting
+        if self.company_id.anglo_saxon_accounting and invoice.type == 'in_invoice':
+            fiscal_position = self.pool.get('account.fiscal.position')
+            for _, _, line_dict in invoice_data['invoice_line']:
```

Comment:
use dummy, dummy, line_dict instead of _, _, line_dict please, as _ could be used sooner or later for translation